bin/GoogleAdsCleanupServices.php

```diff
@@ -0,0 +1,529 @@
+		'KeywordPlanIdea',
+		'KeywordPlan',
+		'KeywordThemeConstant',
+		'KeywordView',
+		'Label',
+		'LandingPageView',
+		'LanguageConstant',
+		'LifeEvent',
+		'LocationView',
+		'ManagedPlacementView',
+		'MediaFile',
+		'MobileAppCategoryConstant',
+		'MobileDeviceConstant',
+		'OfflineUserDataJob',
+		'OperatingSystemVersionConstant',
+		'PaidOrganicSearchTermView',
+		'ParentalStatusView',
+		'PaymentsAccount',
+		'ProductBiddingCategoryConstant',
+		'ProductGroupView',
+		'ReachPlan',
+		'Recommendation',
+		'RemarketingAction',
+		'SearchTermView',
+		'SharedCriterion',
+		'SharedSet',
+		'ShoppingPerformanceView',
+		'SmartCampaignSearchTermView',
+		'SmartCampaignSetting',
+		'SmartCampaignSuggest',
+		'ThirdPartyAppAnalyticsLink',
+		'TopicConstant',
+		'TopicView',
+		'UserData',
+		'UserInterest',
+		'UserList',
+		'UserLocationView',
+		'Video',
+		'WebpageView',
+	];
+
+	/**
+	 * API version.
+	 *
+	 * @var string
+	 */
+	protected $version = 'V9';
+
+	/**
+	 * @var Event Composer event.
+	 */
+	protected $event = null;
+
+	/**
+	 * @var string The path of the Ads library.
+	 */
+	protected $path = null;
+
+	/**
+	 * @var string Source path.
+	 */
+	protected $src_path = '/src/Google/Ads/GoogleAds';
+
+	/**
+	 * @var string Source code path.
+	 */
+	protected $code_path = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Event|null  $event Composer event.
+	 * @param string|null $path  Path of the Ads library.
+	 */
+	public function __construct( Event $event = null, string $path = null ) {
+		$this->event     = $event;
+		$this->path      = $path ?: dirname( __DIR__ ) . '/vendor/googleads/google-ads-php';
+		$this->code_path = dirname( __DIR__ ) . '/src';
+	}
+
+	/**
+	 * Remove services.
+	 *
+	 * @param Event $event Event context provided by Composer
+	 */
+	public static function remove( Event $event = null ) {
+		$cleanup = new GoogleAdsCleanupServices( $event );
+		$cleanup->remove_services();
+		$cleanup->remove_enums();
+	}
+
+	/**
+	 * Remove all services.
+	 */
+	protected function remove_services() {
+		$this->output_text( 'Removing unused services from Google Ads library ' . $this->version );
+
+		foreach ( $this->services as $service ) {
+			$this->remove_service( $service );
+		}
+	}
+
+	/**
+	 * Remove a specific service.
+	 *
+	 * @param string $service Service name.
+	 */
+	protected function remove_service( string $service ) {
+		$this->output_text( "Removing service {$service}" );
+		$this->src_path = '/src/Google/Ads/GoogleAds';
+
+		$file = "/Util/{$this->version}/ResourceNames.php";
+		$this->remove_use_statement( $file, "{$service}ServiceClient" );
+		$this->remove_function( $file, "for{$service}" );
+
+		$file = "/Lib/{$this->version}/ServiceClientFactoryTrait.php";
+		$this->remove_use_statement( $file, "{$service}ServiceClient" );
+		$this->remove_function( $file, "get{$service}ServiceClient" );
+
+		$this->remove_file( "/{$this->version}/Services/Gapic/{$service}ServiceGapicClient.php" );
```

Comment:
💅 

Maybe I misses some of them but I guess `this->version` is added in every `remove_file` function call. HEnce I guess we can just added it there.

Reply:
I decided not to implement this change, while we technically can move the version to the `remove_file` function it would make the function inconsistent with the behaviour of `remove_function` / `remove_use_statement`.

Based on the folder structure we can sometimes have `Lib/<version>...` and other times `<version>/Enums...` so I prefer to be consistent in the file string we pass to the functions.